### PR TITLE
fix(background): better error message for `connectWallet`

### PR DESCRIPTION
**Pull request by sidvishnoi**
## Context

Part of https://github.com/interledger/web-monetization-extension/issues/147

### CODE REVIEW

**Comment by raducristianpopa** [comment]
<h2>Extension builds preview</h2>

<table role="table">
<thead>
  <tr>
    <th>Name</th>
    <th>Link</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td align="center">Latest commit</td>
    <td align="center">03f7e72cef9ad444edcb48c9af4ce740adc5a039</td>
  </tr>
  <tr>
    <td align="center">Latest job logs</td>
    <td align="center"><a href="https://github.com/interledger/web-monetization-extension/actions/runs/9807877104">Run #9807877104</a></td>
  </tr>
  <tr><td align="center">Badge</td><td align="center">N/A</td></tr><tr><td align="center">Badge</td><td align="center">N/A</td></tr>
</tbody>
</table>

File: tsconfig.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "baseUrl": "./src",
     "esModuleInterop": true,
     "module": "commonjs",
-    "target": "ES2020",
+    "target": "ES2022",
```
Comment (sidvishnoi) [inline tsconfig.json:9]: For `Error.cause`

File: src/_locales/en/messages.json [hunk 2]
```diff
@@ -23,5 +23,8 @@
   },
   "siteNotMonetized": {
     "message": "This website is not monetized."
+  },
+  "error_connectWallet_invalidClient": {
+    "message": "Failed to connect. Did you copy the public key to the right wallet address?"
```
Comment (sidvishnoi) [inline src/_locales/en/messages.json:28]: Should it say "right wallet and address"?

**Review by dianafulga — APPROVED** [review]

File: src/_locales/en/messages.json (hunk 2 above)
Comment (raducristianpopa) [reply src/_locales/en/messages.json:28]: As is, It's the right error message. The key is bound to the wallet address, not the wallet.

Comment (raducristianpopa) [inline src/_locales/en/messages.json:28]:
```suggestion
    "message": "Failed to connect. Did you copy the public key to the correct wallet address?"
```

File: src/_locales/en/messages.json [hunk 3]
```diff
@@ -23,5 +23,8 @@
   },
   "siteNotMonetized": {
     "message": "This website is not monetized."
+  },
+  "error_connectWallet_invalidClient": {
```
[the remark this answers is not in the document]
Comment (sidvishnoi) [reply src/_locales/en/messages.json:27]: I was not going for openPayments here as this error is specific to one we show during connect wallet. When it happens during payment (openPayments_invalidClient again), we go for "key revoked" state which has its own error messages. We might show a different one during reconnectWallet even.

**Review by raducristianpopa — APPROVED** [review]

**Pull request merged**